File: frontend/src/services/chatgptService.ts
import api, { ApiResponse } from './api';
import {
  GenerateIdeasRequest,
  GenerateIdeasResponse,
  GenerateSummaryRequest,
  GenerateSummaryResponse,
  GenerateAnnotationRequest,
  GenerateAnnotationResponse,
  ChatGPTInteraction,
} from '../types';

export const chatgptService = {
  async generateIdeas(data: GenerateIdeasRequest): Promise<GenerateIdeasResponse> {
    const response = await api.post<ApiResponse<GenerateIdeasResponse>>('/chatgpt/generate-ideas', data);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to generate ideas');
    }
    return response.data.data;
  },

  async generateSummary(data: GenerateSummaryRequest): Promise<GenerateSummaryResponse> {
    const response = await api.post<ApiResponse<GenerateSummaryResponse>>('/chatgpt/generate-summary', data);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to generate summary');
    }
    return response.data.data;
  },

  async generateAnnotation(data: GenerateAnnotationRequest): Promise<GenerateAnnotationResponse> {
    const response = await api.post<ApiResponse<GenerateAnnotationResponse>>('/chatgpt/generate-annotation', data);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to generate annotation');
    }
    return response.data.data;
  },

  async getSessionInteractions(sessionId: string): Promise<ChatGPTInteraction[]> {
    const response = await api.get<ApiResponse<ChatGPTInteraction[]>>(`/chatgpt/session/${sessionId}/interactions`);
    return response.data.data || [];
  },

  async getMyInteractions(): Promise<ChatGPTInteraction[]> {
    const response = await api.get<ApiResponse<ChatGPTInteraction[]>>('/chatgpt/my-interactions');
    return response.data.data || [];
  },

  async checkRateLimit(sessionId: string): Promise<{ canRequest: boolean; usedCount: number; maxAllowed: number }> {
    const response = await api.get<ApiResponse<{ canRequest: boolean; usedCount: number; maxAllowed: number }>>(`/chatgpt/session/${sessionId}/rate-limit`);
    if (!response.data.data) {
      throw new Error(response.data.message || 'Failed to check rate limit');
    }
    return response.data.data;
  },
};

File: mobile/lib/services/chatgpt_service.dart
import '../core/constants/api_constants.dart';
import 'api_client.dart';

class ChatGPTService {
  final ApiClient _apiClient = ApiClient();

  /// Generate AI ideas for a topic
  Future<List<String>> generateIdeas({
    required String sessionId,
    required String topicDescription,
    int count = 3,
  }) async {
    final response = await _apiClient.post(
      '${ApiConstants.chatgpt}/generate-ideas',
      data: {
        'sessionId': sessionId,
        'topicDescription': topicDescription,
        'count': count,
      },
    );
    final List<dynamic> ideas = response.data['generatedIdeas'] ?? [];
    return ideas.map((e) => e.toString()).toList();
  }

  /// Generate session summary
  Future<String> generateSummary(String sessionId) async {
    final response = await _apiClient.post(
      '${ApiConstants.chatgpt}/generate-summary',
      data: {'sessionId': sessionId},
    );
    return response.data['summary'] ?? '';
  }

  /// Generate annotation for an idea
  Future<String> generateAnnotation({
    required String ideaId,
    required String ideaContent,
    required String topicDescription,
  }) async {
    final response = await _apiClient.post(
      '${ApiConstants.chatgpt}/generate-annotation',
      data: {
        'ideaId': ideaId,
        'ideaContent': ideaContent,
        'topicDescription': topicDescription,
      },
    );
    return response.data['annotation'] ?? '';
  }

  /// Check rate limit for AI requests
  Future<Map<String, dynamic>> checkRateLimit(String sessionId) async {
    final response = await _apiClient.get(
      '${ApiConstants.chatgpt}/session/$sessionId/rate-limit',
    );
    return {
      'canRequest': response.data['canRequest'] ?? false,
      'usedCount': response.data['usedCount'] ?? 0,
      'maxAllowed': response.data['maxAllowed'] ?? 10,
    };
  }
}
